### PR TITLE
Changed parallel universe HF to Cancun HF and some important fixes

### PR DESCRIPTION
**Pull request by pratikspatil024**
# Description

Changed parallel universe HF to Napoli HF

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] Changes only for a subset of nodes

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Nodes audience

In case this PR includes changes that must be applied only to a subset of nodes, please specify how you handled it (e.g. by adding a flag with a default value...)

# Checklist

- [ ] I have added at least 2 reviewer or the whole pos-v1 team
- [ ] I have added sufficient documentation in code
- [ ] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply
- [ ] Created a task in Jira and informed the team for implementation in Erigon client (if applicable)
- [ ] Includes RPC methods changes, and the Notion documentation has been updated

# Cross repository changes

- [ ] This PR requires changes to heimdall
    - In case link the PR here:
- [ ] This PR requires changes to matic-cli
    - In case link the PR here:

## Testing

- [ ] I have added unit tests
- [ ] I have added tests to CI
- [ ] I have tested this code manually on local environment
- [ ] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai
- [ ] I have created new e2e tests into express-cli

### Manual tests

Please complete this section with the steps you performed if you ran manual tests for this functionality, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

File: params/config.go [hunk 1]
```diff
@@ -567,7 +567,7 @@ type BorConfig struct {
 	BurntContract              map[string]string      `json:"burntContract"`              // governance contract where the token will be sent to and burnt in london fork
 	JaipurBlock                *big.Int               `json:"jaipurBlock"`                // Jaipur switch block (nil = no fork, 0 = already on jaipur)
 	DelhiBlock                 *big.Int               `json:"delhiBlock"`                 // Delhi switch block (nil = no fork, 0 = already on delhi)
-	ParallelUniverseBlock      *big.Int               `json:"parallelUniverseBlock"`      // TODO: update all occurrence, change name and finalize number (hardfork for block-stm related changes)
+	NapoliBlock                *big.Int               `json:"napoliBlock"`                // TODO: update all occurrence, change name and finalize number (hardfork for block-stm related changes)
```
Comment (marcello33) [inline params/config.go:570]: Minor one: this won't only be block-stm related HF, right?

**Comment by marcello33** [comment]
I think e2e tests fail because of your branch (and the upper one v1.2.2-beta-candidate still using the old node version.
See [here](https://github.com/maticnetwork/bor/blob/arpit/v1.2.2-beta-candidate/.github/workflows/ci.yml#L220).
This has been updated in `develop` and `master` branch.

**Comment by temaniarpit27** [comment]
> I think e2e tests fail because of your branch (and the upper one `v1.2.2-beta-candidate` still using the old `node` version. See [here](https://github.com/maticnetwork/bor/blob/arpit/v1.2.2-beta-candidate/.github/workflows/ci.yml#L220). This has been updated in `develop` and `master` branch.

@pratikspatil024 please cherry-pick those changes in this branch

File: params/config.go (hunk 1 above)
Comment (pratikspatil024) [reply params/config.go:570]: Yeah, sorry, my bad. Discussed with @temaniarpit27, this will be Cancun Block in bor. In Erigon it will be called Napoli.

**Review by temaniarpit27 — APPROVED** [review]

**Review by manav2401 — APPROVED** [review]

File: core/parallel_state_processor.go [hunk 2]
```diff
@@ -288,6 +288,10 @@ func (p *ParallelStateProcessor) Process(block *types.Block, statedb *state.Stat
 
 	deps := GetDeps(blockTxDependency)
 
+	if !VerifyDeps(deps) {
```
Comment (cffls) [inline core/parallel_state_processor.go:291]: Let's verify the length of `deps` is equal to the length of transactions in this block

Comment (pratikspatil024) [reply core/parallel_state_processor.go]: [Done](https://github.com/maticnetwork/bor/pull/1132/commits/60b5c4143205586278f0047fe1d4784d56254a6b), thanks!

**Review by cffls — APPROVED** [review]

**Pull request merged**